Amplify/Categories/DataStore/Model/Model+DateFormatting.swift

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright 2018-2019 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public struct ModelDateFormatting {
+    static let iso8601WithFractionalSeconds: ISO8601DateFormatter = {
+        let formatter = ISO8601DateFormatter()
+        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
+        return formatter
+    }()
+
+    static let iso8601WithoutFractionalSeconds: ISO8601DateFormatter = {
+        let formatter = ISO8601DateFormatter()
+        formatter.formatOptions = [.withInternetDateTime]
+        return formatter
+    }()
+
+    public static let decodingStrategy: JSONDecoder.DateDecodingStrategy = {
+        let strategy = JSONDecoder.DateDecodingStrategy.custom { decoder -> Date in
+            let container = try decoder.singleValueContainer()
+            let dateString = try container.decode(String.self)
+
+            if let date = iso8601WithFractionalSeconds.date(from: dateString) {
+                return date
+            }
+
+            if let date = iso8601WithoutFractionalSeconds.date(from: dateString) {
+                return date
+            }
+
+            return try container.decode(Date.self)
+        }
+
+        return strategy
+    }()
+
+    public static let encodingStrategy: JSONEncoder.DateEncodingStrategy = {
+        let strategy = JSONEncoder.DateEncodingStrategy.custom { date, encoder in
+            let dateString = iso8601WithFractionalSeconds.string(from: date)
```

Comment:
Hello, getting to this commit trying to understand an issue I have found. It seems `AWSDate` cannot be used in a schema... but I am not sure if this commit is the source of it ([reported here](https://github.com/aws-amplify/amplify-ios/issues/278)) but I would love to understand how this works in case I can contribute in the future :|